code/modules/economy/cargo_telepads.dm

```diff
@@ -0,0 +1,28 @@
+/obj/machinery/telepad_cargo
+	name = "cargo telepad"
+	desc = "A telepad used to recieve imports and send exports."
+	icon = 'icons/obj/machines/telepad.dmi'
+	icon_state = "pad-idle"
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 20
+	active_power_usage = 500
+	var/telepad_id = 0
+/obj/machinery/telepad_cargo/New()
+	telepad_id = random_id(type,10000,99999)
+
+/obj/machinery/telepad_cargo/attackby(obj/item/O as obj, mob/user as mob, params)
+	if(IS_WRENCH(O))
+		playsound(src, 'sound/items/Ratchet.ogg', 50, 1)
+		if(anchored)
+			anchored = 0
+			to_chat(user, "<span class = 'caution'> The [src] can now be moved.</span>")
+		else if(!anchored)
+			anchored = 1
+			to_chat(user, "<span class = 'caution'> The [src] is now secured.</span>")
+	if(IS_MULTITOOL(O))
```

Comment:
I suggest no doing it like that, because you'll be fighting with the base class stuff. They've generalized multitool interactions so you don't need to intercept it in attackby anymore.

You might want to take a look at the multitool extension ``/datum/extension/interactive/multitool`` it allows you to define things that using a multitool on something can do.

If you want an example, I suggested looking at ``code\game\machinery\embedded_controller\airlock_controllers_dummy.dm`` line 108. There's an implementation for using a multitool to setup connections between airlock machines.

Reply:
I've looked into the multitool extensions and that would be appropriate if I needed a full UI for this, but because it's calling a simple input and changing a single variable, the current code is as simple, fast and readable as it can be.

Reply:
Alright then.